src/main/scala/mesosphere/marathon/stream/Repeater.scala

```diff
@@ -0,0 +1,91 @@
+package mesosphere.marathon
+package stream
+
+import akka.actor.Cancellable
+import akka.stream.scaladsl.{ Sink => AkkaSink }
+import akka.Done
+import akka.stream.scaladsl.{ Source, SourceQueueWithComplete }
+import akka.stream.OverflowStrategy
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success, Try }
+
+object Repeater {
+  private class RepeaterLogic[T](queueSize: Int, overflowStrategy: OverflowStrategy)(implicit ec: ExecutionContext) {
+    @volatile private var subscribers: Set[SourceQueueWithComplete[T]] = Set.empty
+    @volatile private var lastElement: T = _
+    @volatile private var elementPushed = false
+    @volatile private var completionResult: Option[Try[Done]] = None
+
+    val onElement: T => Unit = { t =>
+      synchronized {
+        elementPushed = true
+        lastElement = t
+        subscribers.foreach(_.offer(t))
+      }
+    }
+
+    val newSubscriberSource = Source.queue[T](queueSize, overflowStrategy).mapMaterializedValue { sq =>
```

Comment:
Unfortunately `Source.queue` is single threaded thing. What do you think about using BroadcastHub and attaching buffer with a drop strategy?

Reply:
Is the documentation insufficient for Repeater? I have a very good reason for not using BroadcastHub.

As for `Source.queue` being a single threaded thing, this is not how it works :) I'm happy to discuss with you. You can feed elements concurrently to `Source.queue`. The states are either serial (map) and concurrent (mapAsync).